### PR TITLE
Closes #8431: Notify all new observers about change on enableTrackingProtection

### LINKED ISSUE #8431: SpeculativeSessionObserver is swallowing events meant to be for EngineObserver
In https://github.com/mozilla-mobile/android-components/commit/68559df8c0aa4ec388490a3eb143eee634d0c364, we introduced [SpeculativeSessionObserver](https://github.com/mozilla-mobile/android-components/blob/a1a5c582c28386a55f9dcc50f10f4a3011dd3742/components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/util/SpeculativeSessionFactory.kt#L145)  with the goal of capturing early crashes but as a side effect it's consuming early events that were meant to be consumed by the [EngineObserver](https://github.com/mozilla-mobile/android-components/blob/a1a5c582c28386a55f9dcc50f10f4a3011dd3742/components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt#L48), that it's the case of [EngineObserver#onTrackerBlockingEnabledChange](https://github.com/mozilla-mobile/android-components/blob/a1a5c582c28386a55f9dcc50f10f4a3011dd3742/components/browser/session/src/main/java/mozilla/components/browser/session/engine/EngineObserver.kt#L175) causing issues like https://github.com/mozilla-mobile/fenix/issues/15152 where the `EngineObserver` got `onTrackerBlockingEnabledChange` as the `SpeculativeSessionObserver`  consumed it before hand.

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ permalink: /changelog/
 * [Gecko](https://github.com/mozilla-mobile/android-components/blob/master/buildSrc/src/main/java/Gecko.kt)
 * [Configuration](https://github.com/mozilla-mobile/android-components/blob/master/.config.yml)
 
+* **browser-engine-gecko**, **browser-engine-gecko-beta**, **browser-engine-gecko-nightly**
+  * 🚒 Bug fixed [issue #8431](https://github.com/mozilla-mobile/android-components/issues/8431) SpeculativeSessionObserver is swallowing events meant to be for EngineObserver.
```
Comment (pocmo) [inline docs/changelog.md:16]: nit: This is describing the internal detail. Can we describe what it fixes from the point of the view of the consumer? :)

**Review by pocmo — APPROVED** [review]

File: docs/changelog.md (hunk 1 above)
Comment (Amejia481) [reply docs/changelog.md:16]: Should you be 😴? 🤔

Comment (Amejia481) [reply docs/changelog.md:16]: Thanks, anyways :)

**Comment by Amejia481** [comment]
bors r=pocmo

**Pull request merged**